File: modulo8/ejercicio1.py

def flag_printer(value):
    flag = ''

    for k in range(value):
        for i in range(value):
            if k == i or i + k == value - 1:
                flag += '#'
            else:
                flag += '0'

        flag += '\n'

    print(flag)


def main():
    while True:
        ld = int(input())
        if ld != 0:
            flag_printer(ld)
        else:
            break


if __name__ == '__main__':
    main()

File: modulo8/ejercicio1.rs
use std::io;

fn flag_printer(value: u32) {
    let mut flag = String::from("");

    for k in 0..value {
        for i in 0..value {
            if k == i || (i + k) == value - 1 {
                flag.push_str("#");
            } else {
                flag.push_str("0");
            }
        }

        flag.push_str("\n")
    }

    println!("{}", flag);
}

fn main() {
    loop {
        let mut ld = String::new();
        io::stdin()
            .read_line(&mut ld)
            .expect("Failed to read from stdin");

        let ld: u32 = match ld.trim().parse() {
            Ok(num) => num,
            Err(_) => {
                println!("Type a number!");
                continue;
            }
        };

        if ld != 0 {
            flag_printer(ld);
        } else {
            break;
        }
    }
}
